OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyUtils.cs

```diff
@@ -551,7 +555,10 @@ public static void SetupColorWidget(Widget parent, Session.Client c)
 		public static void SetupFactionWidget(Widget parent, Session.Client c, Dictionary<string, LobbyFaction> factions)
 		{
 			var factionName = parent.Get<LabelWidget>("FACTIONNAME");
-			factionName.GetText = () => factions[c.Faction].Name;
+			var truncated = new CachedTransform<string, string>(name =>
+				WidgetUtils.TruncateText(name, factionName.Bounds.Width, Game.Renderer.Fonts[factionName.Font]));
+			factionName.GetText = () => truncated.Update(factions[c.Faction].Name);
```

Comment:
Ideally this takes `c.Faction` as parameter and does the array lookup inside the function for the cached transform.